### PR TITLE
fix: transaction calls on the reth-provider should not generate a hash by default

### PR DESCRIPTION
**Pull request by joshstevens19**
When calling the transaction methods by default, it should not compute the tx hash (`TransactionSigned`) as this dramatically slows down the response when using it and means it computes the hash even if you do not need to use it; you have the .hash() method on the `TransactionSignedNoHash` to do this. 

(was not sure where i run `cargo fmt` as i did that and a lot of file changes so just assumed the CI may do it let me know if not)

### CODE REVIEW

**Comment by joshstevens19** [comment]
found due to migrating to the provider and it made the results slower > https://github.com/joshstevens19/reth-indexer/pull/15

File: crates/storage/provider/src/providers/database/mod.rs [hunk 1]
```diff
@@ -224,18 +224,18 @@ impl<DB: Database> TransactionsProvider for ProviderFactory<DB> {
         self.provider()?.transaction_id(tx_hash)
     }
 
-    fn transaction_by_id(&self, id: TxNumber) -> Result<Option<TransactionSigned>> {
+    fn transaction_by_id(&self, id: TxNumber) -> Result<Option<TransactionSignedNoHash>> {
         self.provider()?.transaction_by_id(id)
     }
 
-    fn transaction_by_hash(&self, hash: TxHash) -> Result<Option<TransactionSigned>> {
+    fn transaction_by_hash(&self, hash: TxHash) -> Result<Option<TransactionSignedNoHash>> {
```
Comment (mattsse) [inline crates/storage/provider/src/providers/database/mod.rs:231]: I can see how this is useful for perf

I'd rather have this as separate trait functions like `transaction_by_id_no_hash` instead and keep returning `TransactionSigned`

**Review by mattsse — CHANGES_REQUESTED** [review]

File: crates/storage/provider/src/providers/database/mod.rs (hunk 1 above)
Comment (joshstevens19) [reply crates/storage/provider/src/providers/database/mod.rs:231]: Cool il do that then less changes and fine my end 👍

**Review by mattsse — CHANGES_REQUESTED** [review]
lgtm, pending CI

I think you're missing some impls

File: crates/storage/provider/src/providers/database/mod.rs (hunk 1 above)
Comment (joshieDo) [reply crates/storage/provider/src/providers/database/mod.rs]: i'm not really sure about it though, it hides the performance cost. Can we just add to the doc of the `databaseprovider` implementation, that it incurs that cost at least?

Comment (joshstevens19) [reply crates/storage/provider/src/providers/database/mod.rs]: let me know what you guys want to do.. @mattsse @joshieDo

Comment (mattsse) [reply crates/storage/provider/src/providers/database/mod.rs]: > 'm not really sure about it though

about what exactly?

Comment (joshstevens19) [reply crates/storage/provider/src/providers/database/mod.rs]: added a note on it

**Comment by joshstevens19** [comment]
@mattsse ready to merge

**Review by joshieDo — APPROVED** [review]

File: crates/storage/provider/src/providers/database/mod.rs (hunk 1 above)
Comment (mattsse) [reply crates/storage/provider/src/providers/database/mod.rs]: > about returning TransactionSigned per default on functions that need to calculate the hash. the calculation cost is kinda hidden imo

right, I was strictly looking at this from rpc pov where we always need this

**Review by mattsse — APPROVED** [review]

**Pull request merged**